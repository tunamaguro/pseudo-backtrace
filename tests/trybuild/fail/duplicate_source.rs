use pseudo_backtrace::StackError;

#[track_caller]
fn location() -> &'static core::panic::Location<'static> {
    core::panic::Location::caller()
}

#[derive(Debug, StackError)]
struct DuplicateSource {
    #[source]
    first: std::io::Error,
    #[source]
    second: std::io::Error,
    #[location]
    location: &'static core::panic::Location<'static>,
}

impl DuplicateSource {
    pub fn new() -> Self {
        Self {
            first: std::io::Error::from_raw_os_error(5),
            second: std::io::Error::from_raw_os_error(6),
            location: location(),
        }
    }
}

impl core::fmt::Display for DuplicateSource {
    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
        write!(f, "dup-source")
    }
}

impl core::error::Error for DuplicateSource {}

fn main() {}
