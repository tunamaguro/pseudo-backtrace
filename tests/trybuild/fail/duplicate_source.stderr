error: duplicate `#[source]` or `#[stack_error]` attribute
  --> tests/trybuild/fail/duplicate_source.rs:12:5
   |
12 | /     #[source]
13 | |     second: std::io::Error,
   | |__________________________^
